src/lab8.py

```diff
@@ -279,12 +279,14 @@ def render(self):
         paint_tree(self.document, self.display_list)
 
     def click(self, x, y):
+        if self.focus:
+            self.focus.is_focused = False
         self.focus = None
         y += self.scroll
         objs = [obj for obj in tree_to_list(self.document, [])
                 if obj.x <= x < obj.x + obj.width
                 and obj.y <= y < obj.y + obj.height]
-        if not objs: return
+        if not objs: return self.render()
```

Comment:
I think this is the core of the bug fix, and it's a good catch.

Reply:
Yes, but also cases where the input element is not in the ancestor chain?